---
title: "3_Genome_Alignment and concensus mapping"
output: html_document
---

Next up, we want to align the cm21 genome to the af293 reference genome. I will be using bwa for aligning

First, the reference genome is indexed so it can by bwa mem. I have included that within the bash script. It should be done separately as it may take some time.


```{bash}
#!/bin/bash
#SBATCH --account=def-jpxu
#SBATCH --time=1:30:00
#SBATCH --cpus-per-task=16
#SBATCH --nodes=2
#SBATCH --mem=60G
export OMP_NUM_THREADS=$SLURM_CPUS_PER_TASK

module load bwa
module load samtools
module load picard

bwa index ~/722_project/ref_Af293.fa

bwa mem -t 8 ~/722_project/ref_Af293.fa $1 $2  | samtools view -bSh > ~/scratch/cm21.bam

samtools sort ~/scratch/cm21.bam -o ~/scratch/cm21_sorted.bam

samtools index cm21_sorted.bam

samtools flagstat cm21_sorted.bam

java -jar $EBROOTPICARD/picard.jar MarkDuplicates I=cm21_sorted.bam O=cm21.markdup.bam M=cm21_mark_metrics.txt

samtools index cm21.markdup.bam

samtools flagstat cm21.markdup.bam > cm21.markdup_mstats.txt



sacct -j $SLURM_JOB_ID --format=JobID%16,Submit,Start,Elapsed,NCPUS,ExitCode,NodeList%8

```

Instead of the java line, I attempted to use samtools markdup. This command requires samtool fixmate to be run prior, but I was unable to get it to work with my data. I therefore elected to use picard to mark dupicates

```{bash}
cat ~/722_project/ref_Af293.fa | vcf-consensus cm21SNPs.vcf.gz > cm21_con.fa
```
