README page






Welcome to my 722 project Github page. For my project, I looked to identify and investigate microsatellite loci from reference and a lab strain of A.fumigatus


In this project, I used Sharcnet to conduct my analyses. I jumped between the two as I could not get the PERF program to work from within my home directory. Therefore I moved to Sharcnet to get that aspect of my project done. I detatiled waht PERF does below.




I downloaded the SRR file 
```{bash}
wget https://sra-download.ncbi.nlm.nih.gov/traces/sra50/SRR/012591/SRR12894203
```



```{bash}
module load fastqc
mkdir ~/scratch/fastqc_outputs 
nano fastqc.sh
```
```{bash}
fastqc.sh script

#!/bin/bash
#SBATCH --account=def-jpxu
#SBATCH --time=00:30:00
#SBATCH --cpus-per-task=8
#SBATCH --nodes=2
#SBATCH --mem=8G
export OMP_NUM_THREADS=$SLURM_CPUS_PER_TASK

fastqc *.fastq -o ~/scratch/fastqc_outputs/

```




Installing the dependencies required for PERF to run
```{bash}
pip install biopython --no-index
pip install regex --no-index
pip install tqdm --no-index
```


Once downloaded from NCBI, the file format was changed to fastq.

Need to open up the sra_toolkit on shracnet
```{bash}
module load StdEnv/2020
module load gcc/9.3.0

```


```{bash}
#!/bin/bash
#SBATCH --account=def-jpxu
#SBATCH --time=01:15:00
#SBATCH --cpus-per-task=8
#SBATCH —nodes=2
#SBATCH --mem=30G
export OMP_NUM_THREADS=$SLURM_CPUS_PER_TASK

fasterq-dump --split-files $1
```

Make it an executable
```{bash}
chmod +x SRR_fastq.sh
```


A CM21 strain
```{bash}
/usr/local/sratoolkit/fastq-dump.2.8.2 --split-files SRR12894203
```


Testing the reference strain for Microsats



I also tried to identify where each repeat was located. this includes Wheter it was intergenic, or within introns, exons, etc.
The GFF file with annotation details for af293. Download it from NCBI and unzip it using tar
This gave me trouble to get it to work. I was not able to identify what was causing the problem. I tried using both GTF and GFF files, however neither worked. most errors appeared as below 
```{bash}
[gk1995@gra-login1 722_project]$ ~/bash_files/perf.sh ref_Af293.fa GTF_test af293.GFF gff
Using length cutoff of 12
Processing NC_007201.1: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8/8 [00:15<00:00,  1.93s/it]

Generating annotations for identified repeats..

  0%|                                                                                                                                                                                                           | 0/21138 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/home/gk1995/.local/bin/PERF", line 8, in <module>
    sys.exit(main())
  File "/home/gk1995/.local/lib/python3.8/site-packages/PERF/core.py", line 162, in main
    ssr_native(args, length_cutoff=args.min_length)
  File "/home/gk1995/.local/lib/python3.8/site-packages/PERF/core.py", line 106, in ssr_native
    fasta_ssrs(args, repeats_info)
  File "/home/gk1995/.local/lib/python3.8/site-packages/PERF/rep_utils.py", line 253, in fasta_ssrs
    annotate(args)
  File "/home/gk1995/.local/lib/python3.8/site-packages/PERF/annotation.py", line 291, in annotate
    minStartIndex += minIndex
UnboundLocalError: local variable 'minIndex' referenced before assignment
```

A more informative error occured when I ran the GFF file. I updated my script as follows

```{bash}
[gk1995@gra-login1 722_project]$ ~/bash_files/perf.sh ref_Af293.fa GTF_test af293.GFF gff ID

Using length cutoff of 12
Processing NC_007201.1: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8/8 [00:15<00:00,  1.95s/it]

GeneKeyError:
The attribute "gene" is not among the attributes for gene. Please select a different one.
The available ones are [ID, Name, end_range, gbkey, gene_biotype, locus_tag, old_locus_tag, partial, start_range]

```

so I added the --anno-format flag to specify GFF file and the --gene-key flag to specify "ID" shown in the above error and what is seen in the GFF file

```{bash}
perf.sh script

#!/bin/bash
#SBATCH --account=def-jpxu
#SBATCH --time=00:10:00
#SBATCH --cpus-per-task=6
#SBATCH —nodes=2
#SBATCH --mem=15G
export OMP_NUM_THREADS=$SLURM_CPUS_PER_TASK

PERF -input $1 --output $2 --anotate $3 --anno-format $4 --gene-key $5 --analyse

```

GFF run with ID stated. Results in a error in the PERF script and does not generate the SSR anotation file.
```{bash}
[gk1995@gra-login1 722_project]$ ~/bash_files/perf.sh ref_Af293.fa GFF_test af293.GFF gff ID

Using length cutoff of 12
Processing NC_007201.1: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8/8 [00:14<00:00,  1.85s/it]

Generating annotations for identified repeats..

  0%|                                                                                                                                                                                                           | 0/21138 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/home/gk1995/.local/bin/PERF", line 8, in <module>
    sys.exit(main())
  File "/home/gk1995/.local/lib/python3.8/site-packages/PERF/core.py", line 162, in main
    ssr_native(args, length_cutoff=args.min_length)
  File "/home/gk1995/.local/lib/python3.8/site-packages/PERF/core.py", line 106, in ssr_native
    fasta_ssrs(args, repeats_info)
  File "/home/gk1995/.local/lib/python3.8/site-packages/PERF/rep_utils.py", line 253, in fasta_ssrs
    annotate(args)
  File "/home/gk1995/.local/lib/python3.8/site-packages/PERF/annotation.py", line 291, in annotate
    minStartIndex += minIndex
UnboundLocalError: local variable 'minIndex' referenced before assignment

```




I ran the code
```{bash}
[gk1995@gra-login1 722_project]$ ~/bash_files/perf.sh ref_Af293.fa GFF_test af293.GFF gff ID
```
but after the initial SSR call, a similar error occured


Next I tried specifiying the file type and the 

It would have provided me the information below. Table taken from https://raw.githubusercontent.com/RKMlab/perf/master/README.md

| S.No | Column | Description |
|:----:| ------ | ----------- |
| 9 | Gene name | Name of the closest gene |
| 10 | Gene Start | Start position of gene in the Chromosome |
| 11 | Gene Stop | End position of gene in the Chromosome |
| 12 | Strand | The strand orientation of the gene |
| 13 | Genomic annotation | Annotation of the repeat w.r.t to the gene. Possible annotations are {Genic, Exonic, Intronic, Intergenic} |
| 14 | Promoter annotation | If repeat falls in the promoter region of the closest gene. The default promoter region is 1Kb upstream and downstream of TSS. |
| 15 | Distance from TSS | Distance of the repeat from the TSS of the gene. |

```{bash}
tar -xvf genome_assemblies_genome_gff.tar > af293.GFF
gunzip ncbi-genomes-2021-04-28/GCA_000002655.1_ASM265v1_genomic.gff.gz
mv ncbi-genomes-2021-04-28/GCA_000002655.1_ASM265v1_genomic.gff af293.GFF
```

```{bash}
#!/bin/bash
#SBATCH --account=def-jpxu
#SBATCH --time=00:03:00
#SBATCH --cpus-per-task=6
#SBATCH —nodes=2
#SBATCH --mem=15G
export OMP_NUM_THREADS=$SLURM_CPUS_PER_TASK

PERF --input $1 --output $2 --anotate $3 --analyse 

```
```{bash}
~/bash_files/perf.sh ref_Af293.fa perf_af293_12 af293.GFF
```




```{bash}
scp -r gk1995@graham.computecanada.ca:~/722_project/perf_* .
```



Obtaining and using perf (reference to paper here)
This gave me a bit of trouble getting it to work on the cluster




Downloading and preparing the reference genome Af293

wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/655/GCF_000002655.1_ASM265v1/GCF_000002655.1_ASM265v1_genomic.fna.gz


gunzip GCF_000002655.1_ASM265v1_genomic.fna.gz
For simplicity, the name of this reference fasta file will be changed to Af293.fa

mv GCF_000002655.1_ASM265v1_genomic.fna ref_Af293.fa
An index file for the reference genome is then made using bwa. This will be done in advance as we will be using BWA-MEM for read alignment later on

bwa index Af293.fa

